rfd/0207-vnet-ssh.md

```diff
@@ -0,0 +1,392 @@
+MFA or hardware keys.
+
+[Per-session MFA](https://goteleport.com/docs/admin-guides/access-controls/guides/per-session-mfa/)
+requires the client to request and use a new SSH certificate for
+each session, and the OpenSSH client does not know to request and use this
+session certificate.
+Even though the generated OpenSSH config uses a ProxyCommand that calls `tsh` to
+dial the target SSH host, the `tsh` doesn't handle per-session MFA here because
+the third-party SSH client is still ultimately responsible for authenticating to the
+host with the local SSH certificate from disk.
+
+Third-party SSH clients are unable to use
+[hardware keys](https://goteleport.com/docs/admin-guides/access-controls/guides/hardware-key-support/)
+because `tsh` uses a custom file format to reference these keys, which
+third-party clients won't recognize and won't be able to use for signatures.
+
+VNet SSH will simplify access for third-party SSH clients by avoiding the need
+for custom OpenSSH configuration.
+The third-party clients will not need to use any SSH certificate, they will
+connect directly to the local VNet process without any user authentication
+provided by the SSH client.
+The VNet process will "terminate" the incoming SSH connection from the
+third-party client, and dial an outbound SSH connection to the target SSH host,
+then proxy the connection between the two.
+VNet is able to authenticate the user out-of-band from the SSH connection by
+prompting the user from Teleport Connect for any hardware key interactions or
+re-logins necessary.
+Because VNet will own the full SSH connection to the remote host, it can fully
+support per-session MFA and hardware keys.
+
+## Details
+
+### UX
+
+When VNet SSH is active, third-party SSH clients such as OpenSSH will be able
+to connect to teleport SSH hosts at a URI matching
+`<username>@<hostname>.ssh.<proxy-address>`.
+For example, `ssh alice@node.ssh.teleport.example.com` will work to connect as
+the SSH user `alice` to a host named `node` in the cluster
+`teleport.example.com`.
+If per-session MFA is required, Connect will prompt the user to tap their
+hardware key.
+
+VNet SSH will be enabled by default whenever the user runs VNet on their local
+workstation.
+The user will be able to start VNet the same way they do today, by enabling it
+in Connect or running `tsh vnet`.
+
+#### User story - VSCode remote development
+
+Alice is a software engineer with an underpowered laptop and she wants to
+compile and test her application on a more powerful machine.
+She starts a large EC2 instance in her company AWS account for development.
+Her company has an AMI that automatically joins the instance to their company
+Teleport cluster, `teleport.example.com`.
+The machine has the hostname `devbox`.
+
+Alice opens Teleport Connect and sees `devbox` as an SSH server on the main
+resources page.
+Next to the usual "Connect" button on the SSH server card, she sees a new VNet
+button.
```

Comment:
I don't know if we have enough space to put another button there. 😅

We could use this style of button that Sakshyam recently added for SAML apps. We'd have a three dot menu with "Connect with VNet" as the only extra option.
<img width="204" alt="menu-login-with-action-menu" src="https://github.com/user-attachments/assets/196a108d-9d6e-43c1-84c2-fc141b93a5c9" />

Reply:
Yeah I was actually thinking of something like that, i think it's a good idea. would be nice to have _some_ type of button on the server card for discoverability